pkg/controller/eventing/subscription/reconcile.go

```
@@ -0,0 +1,331 @@
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/dynamic"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// Reconcile compares the actual state with the desired, and attempts to
+// converge the two. It then updates the Status block of the Subscription resource
+// with the current status of the resource.
+func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	glog.Infof("Reconciling subscription %v", request)
+	subscription := &v1alpha1.Subscription{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, subscription)
+
+	if errors.IsNotFound(err) {
+		glog.Errorf("could not find subscription %v\n", request)
+		return reconcile.Result{}, nil
+	}
+
+	if err != nil {
+		glog.Errorf("could not fetch Subscription %v for %+v\n", err, request)
+		return reconcile.Result{}, err
+	}
+
+	original := subscription.DeepCopy()
+
+	// Reconcile this copy of the Subscription and then write back any status
+	// updates regardless of whether the reconcile error out.
+	err = r.reconcile(subscription)
+	if equality.Semantic.DeepEqual(original.Status, subscription.Status) {
+		// If we didn't change anything then don't call updateStatus.
+		// This is important because the copy we loaded from the informer's
+		// cache may be stale and we don't want to overwrite a prior update
+		// to status with this stale state.
+	} else if _, err := r.updateStatus(subscription); err != nil {
+		glog.Warningf("Failed to update subscription status: %v", err)
+		return reconcile.Result{}, err
+	}
+
+	// Requeue if the resource is not ready:
+	return reconcile.Result{}, err
+}
+
+func (r *reconciler) reconcile(subscription *v1alpha1.Subscription) error {
+	// See if the subscription has been deleted
+	accessor, err := meta.Accessor(subscription)
+	if err != nil {
+		log.Fatalf("Failed to get metadata: %s", err)
```

I think this should return an error instead of killing the entire process.

---

haha, you and me both.